examples/http-server/Dockerfile

```diff
@@ -1,15 +1,16 @@
-FROM golang:1.22
+# Build stage
+FROM golang:1.22 AS build
 
-RUN mkdir /src/
-WORKDIR /src/
+WORKDIR /src
 COPY . .
 RUN go get ./...
-RUN go build -ldflags "-linkmode external -extldflags -static" -a main.go
+RUN go build -ldflags "-linkmode external -extldflags -static" -a -o /app/main main.go
```

Comment:
Why not build the image in local-system and just copy the build file and the configs - which would reduce the size of the container.

i.e we can use alpine container?

Reply:
Lets discuss the approach on this first before actually doing it

Reply:
@aryanmehrotra IMO Building the app outside the docker is also a good approach which will reduce the docker build time and simplify the docker file.
Building the app inside the docker also have the advantage that we don't need to depend on the host machine for the dependencies. users can simply install docker and they can start the app.
I'm ok with both the approaches

Reply:
seems fine, lets build in the docker itself if needed will change in future